templates/boltzmann/middleware/honeycomb.ts

```diff
@@ -9,77 +10,133 @@ import beeline from 'honeycomb-beeline'
 import isDev from 'are-we-dev'
 void `{% endif %}`;
 
+let otelTracer: OtelTracer | null = null;
+
+function getOtelTracer() {
+  if (!otelTracer) {
+    otelTracer = otelTrace.getTracer('boltzmann', '1.0.0')
+  }
+  return otelTracer
+}
+
 function trace ({
   headerSources = ['x-honeycomb-trace', 'x-request-id'],
 } = {}) {
   if (!process.env.HONEYCOMB_WRITEKEY) {
     return (next: Handler) => (context: Context) => next(context)
   }
 
-  const schema = require('honeycomb-beeline/lib/schema')
-  const tracker = require('honeycomb-beeline/lib/async_tracker')
-
-  return function honeycombTrace (next: Handler) {
-    return (context: Context) => {
-      const traceContext = _getTraceContext(context)
-      const trace = beeline.startTrace({
-        [schema.EVENT_TYPE]: 'boltzmann',
-        [schema.PACKAGE_VERSION]: '1.0.0',
-        [schema.TRACE_SPAN_NAME]: `${context.method} ${context.url.pathname}${context.url.search}`,
-        [schema.TRACE_ID_SOURCE]: traceContext.source,
-        'request.host': context.host,
-        'request.original_url': context.url.href,
-        'request.remote_addr': context.remote,
-        'request.method': context.method,
-        'request.scheme': context.url.protocol,
-        'request.path': context.url.pathname,
-        'request.query': context.url.search
-      },
-      traceContext.traceId,
-      traceContext.parentSpanId,
-      traceContext.dataset)
-
-      if (isDev()) {
-        context._honeycombTrace = trace
-      }
-
-      if (traceContext.customContext) {
-        beeline.addContext(traceContext.customContext)
-      }
+  if (process.env.HONEYCOMB_API_HOST) {
+    return function honeycombOtelTrace (next: Handler) {
+      return (context: Context) => {
+        const activeContext = otelContext.active()
```

Comment:
I still find the active context abstraction beyond confusing, but I *believe* that I'm using it correctly here. The one gap I see is in setting the traceParent and traceState from [the headers](https://w3c.github.io/trace-context/#relationship-between-the-headers), which is something [the CI repo does but for processes](https://github.com/eaze/ci/blob/latest/lib/otel.js#L70-L75). I'm gonna take a crack at pulling these out.

Reply:
OK, this has been updated to pull/set the traceparent and tracestate from the headers, if defined.